src/python2js.c

```diff
@@ -131,6 +131,21 @@ _python2js(PyObject* x, PyObject* map)
     if (x_long == -1 && PyErr_Occurred()) {
       return HW_ERROR;
     }
+    // Since Javascript doesn't support > 32-bit ints, use floats
+    // when the Python int gets too large.  This will lose precision,
+    // but is less problematic than truncation.
+    if ((unsigned long)x_long > 0x7fffffff) {
```

Comment:
Maybe `abs(x_long)`, as if `x_long=-1` wouldn't `(unsigned long)x_long` be 2**63+1 or something above the threshold in any case ? Unless I'm missing something..

Reply:
Indeed, that's correct.  Will fix.